components/SearchForm.tsx

```diff
@@ -1,19 +1,24 @@
-import { FormEvent, useState } from "react";
+import { FormEvent, useRef } from "react";
 import Image from "next/image";
 import { useRouter } from "next/router";
 import styles from "@/components/SearchForm.module.css";
 import search_ic from "@/public/ic_search.svg";
 
 export default function SearchForm() {
   const router = useRouter();
-  const [value, setValue] = useState<string>("");
+  const inputRef = useRef<HTMLInputElement>(null);
 
   const handleFormSubmit = (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault();
 
-    if (!value) return;
+    const value = inputRef.current?.value.trim();
 
-    router.push(`Board?keyword=${value}`);
+    if (!value || /^\s+$/.test(value)) {
```

Comment:
공백 문자로만 구성된 문자열은 trim으로 제거된 상태라 불필요한 조건으로 보이네요.
 
```suggestion
    if (!value) {
```